server/clip_server/model/model.py

```diff
@@ -885,17 +576,91 @@ def load_openclip_model(
                 False
             ), 'pretrained image towers currently only supported for timm models'
 
-    model = CLIP(**model_cfg)
-    model.eval()
-
-    model.load_state_dict(load_state_dict(model_path))
+    cast_dtype = get_cast_dtype(precision)
+    custom_text = (
+        model_cfg.pop('custom_text', False)
+        or force_custom_text
+        or ('hf_model_name' in model_cfg['text_cfg'])
+    )
 
-    if str(device).startswith('cuda'):
-        convert_weights_to_fp16(model)
+    if custom_text:
+        model = CustomTextCLIP(**model_cfg, cast_dtype=cast_dtype)
+    else:
+        model = CLIP(**model_cfg, cast_dtype=cast_dtype)
 
+    model.eval()
+    model.load_state_dict(load_state_dict(model_path))
     model.to(device=device)
 
+    if precision in ("fp16", "bf16"):
+        convert_weights_to_lp(
+            model, dtype=torch.bfloat16 if precision == 'bf16' else torch.float16
+        )
+
     if jit:
         model = torch.jit.script(model)
 
     return model
+
+
+def trace_model(model, batch_size=256, device=torch.device('cpu')):
+    model.eval()
+    image_size = model.visual.image_size
+    example_images = torch.ones((batch_size, 3, image_size, image_size), device=device)
+    example_text = torch.zeros(
+        (batch_size, model.context_length), dtype=torch.int, device=device
+    )
+    model = torch.jit.trace_module(
+        model,
+        inputs=dict(
+            forward=(example_images, example_text),
+            encode_text=(example_text,),
+            encode_image=(example_images,),
+        ),
+    )
+    model.visual.image_size = image_size
+    return model
+
+
+def resize_pos_embed(state_dict, model, interpolation: str = 'bicubic', seq_dim=1):
```

Comment:
Is this one needed?

Reply:
deleted

Reply:
please make sure this one is actually not needed by us

Reply:
no need